pom.xml

```diff
@@ -63,6 +63,10 @@
   </ciManagement>
 
   <properties>
+    <!-- JBoss Parent -->
+    <maven.compiler.target>17</maven.compiler.target>
+    <maven.compiler.source>17</maven.compiler.source>
```

Comment:
Shouldn't this be "maven.compiler.release"?
https://maven.apache.org/plugins/maven-compiler-plugin/examples/set-compiler-source-and-target.html
https://maven.apache.org/plugins/maven-compiler-plugin/examples/set-compiler-release.html
Or does this cause conflicts with the properties defined in jboss-parent?

Reply:
See https://github.com/jboss/jboss-parent-pom/blob/main/pom.xml as it's used in multiple places.

e.g. https://maven.apache.org/plugins/maven-compiler-plugin/examples/set-compiler-source-and-target.html setting doesn't propagate to enforcer.

Reply:
I found https://www.github.com/jboss/jboss-parent-pom/issues/363 with the same discussion to change it in jboss-parent. 

I don't have enough maven knowhow about the details. But it seems we have to set the old properties in order to overwrite the ones defined in the parent pom? The "maven.compiler.release" property has the benefit that the API usage from newer jdk versions would fail to compile even if compilation is done with a newer java version, which would not be checked if "maven.compiler.target" and "maven.compiler.source" properties are set only?

I gave it a try and I can set all three properties and the project still builds. Do you have any experience with this combination?

Reply:
@WolfgangHG you are right, I just assumed that jboss-parent is handling this, but it looks like it does need some updates after all. 

Anyway, combining seems to work as expected, so fixed.

Reply:
A "11" came back with your last change? Shouldn't this be "17" ;-)?

Reply:
Oh no... LOL, fixed now.

Reply:
Thanks, looks good now. Might require further changes, depending on replies to your "jboss-parent" question.

Reply:
Right, but mainly the warnings are done.

[WARNING] location of system modules is not set in conjunction with -source 11
  not setting the location of system modules may lead to class files that cannot run on JDK 11
    --release 11 is recommended instead of -source 11 -target 11 because it sets the location of system modules automatically